sql/connect/server/src/main/scala/org/apache/spark/sql/connect/ui/SparkConnectServerListener.scala

```diff
@@ -80,11 +84,10 @@ private[connect] class SparkConnectServerListener(
       return
     }
     val executeJobTag = executeJobTagOpt.get
-    val exec = executionList.get(executeJobTag)
```

Comment:
This should have been synchronized otherwise it's undefined behavior, e.g., the map could have been resized when reading the entry.